java/ql/lib/semmle/code/java/frameworks/apache/Collections.qll

```diff
@@ -182,6 +204,695 @@ private class ApacheKeyValueModel extends SummaryModelCsv {
+          ".collection;IndexedCollection;true;nonUniqueIndexedCollection;;;Element of Argument[0];Element of ReturnValue;value",
+          ".collection;IndexedCollection;true;get;;;Element of Argument[-1];ReturnValue;value",
+          ".collection;IndexedCollection;true;values;;;Element of Argument[-1];Element of ReturnValue;value",
+          ".collection;PredicatedCollection$Builder;true;add;;;Argument[0];Element of Argument[-1];value",
+          ".collection;PredicatedCollection$Builder;true;addAll;;;Element of Argument[0];Element of Argument[-1];value",
+          ".collection;PredicatedCollection$Builder;true;createPredicatedList;;;Element of Argument[-1];Element of ReturnValue;value",
+          ".collection;PredicatedCollection$Builder;true;createPredicatedList;;;Element of Argument[0];Element of ReturnValue;value",
+          ".collection;PredicatedCollection$Builder;true;createPredicatedSet;;;Element of Argument[-1];Element of ReturnValue;value",
+          ".collection;PredicatedCollection$Builder;true;createPredicatedSet;;;Element of Argument[0];Element of ReturnValue;value",
+          ".collection;PredicatedCollection$Builder;true;createPredicatedMultiSet;;;Element of Argument[-1];Element of ReturnValue;value",
+          ".collection;PredicatedCollection$Builder;true;createPredicatedMultiSet;;;Element of Argument[0];Element of ReturnValue;value",
+          ".collection;PredicatedCollection$Builder;true;createPredicatedMultiBag;;;Element of Argument[-1];Element of ReturnValue;value",
+          ".collection;PredicatedCollection$Builder;true;createPredicatedMultiBag;;;Element of Argument[0];Element of ReturnValue;value",
+          ".collection;PredicatedCollection$Builder;true;createPredicatedMultiQueue;;;Element of Argument[-1];Element of ReturnValue;value",
+          ".collection;PredicatedCollection$Builder;true;createPredicatedMultiQueue;;;Element of Argument[0];Element of ReturnValue;value",
+          ".collection;PredicatedCollection$Builder;true;rejectedElements;;;Element of Argument[-1];Element of ReturnValue;value",
+          ".collection;PredicatedCollection;true;PredicatedCollection;;;Element of Argument[0];Element of Argument[-1];value",
+          ".collection;PredicatedCollection;true;predicatedCollection;;;Element of Argument[0];Element of ReturnValue;value",
+          ".collection;SynchronizedCollection;true;SynchronizedCollection;;;Element of Argument[0];Element of Argument[-1];value",
+          ".collection;SynchronizedCollection;true;synchronizedCollection;;;Element of Argument[0];Element of ReturnValue;value",
+          ".collection;SynchronizedCollection;true;decorated;;;Element of Argument[-1];Element of ReturnValue;value",
+          ".collection;TransformedCollection;true;TransformedCollection;;;Element of Argument[0];Element of Argument[-1];value",
+          ".collection;TransformedCollection;true;transformingCollection;;;Element of Argument[0];Element of ReturnValue;value",
```

Comment:
The docs say "If there are any elements already in the collection being decorated, they will be transformed by this method." So this can't be properly modelled until we have lambda flow.

Reply:
We have lambda flow now.

Reply:
Good point. I think I would prefer to do the extra modelling enabled by lambda flow for this package as a separate task, after this PR has been merged.